yt_dlp/extractor/arte.py

```diff
@@ -46,154 +49,150 @@ class ArteTVIE(ArteTVBaseIE):
                     })
-                formats.extend(m3u8_formats)
+
+                formats.extend(fmts)
+                self._merge_subtitles(subs, target=subtitles)
                 continue
 
-            format = {
-                'format_id': format_id,
-                'language_preference': lang_pref,
-                'format_note': format_note,
-                'width': int_or_none(f.get('width')),
-                'height': int_or_none(f.get('height')),
-                'tbr': int_or_none(f.get('bitrate')),
-                'quality': qfunc(f.get('quality')),
-            }
-
-            if media_type == 'rtmp':
-                format['url'] = f['streamer']
-                format['play_path'] = 'mp4:' + f['url']
-                format['ext'] = 'flv'
-            else:
-                format['url'] = f['url']
-
-            formats.append(format)
-
-        # For this extractor, quality only represents the relative quality
-        # with respect to other formats with the same resolution
-        self._sort_formats(formats, ('res', 'quality'))
+            if stream['protocol'] in ('HTTPS', 'RTMP'):
+                formats.append({
+                    'format_id': stream['protocol'] + '-' + (streamVersionCode or streamVersion.get('shortLabel', 'unknown')),
+                    'url': stream['url'],
+                    'format_note': '[{}] {}'.format(streamVersion.get('shortLabel', '?'), streamVersion.get('label')),
+                    'language_preference': lang_pref,
+                    # 'ext': 'mp4',  # XXX: may or may not be necessary, at least for HTTPS
+                })
+                continue
+
+            self._report_warning(
+                f'Skipping stream with unknown protocol {stream["protocol"]}')
+
+            # XXX: chapters from stream['segments']?
+            # the JS also apparently looks for chapters in
+            # config['data']['attributes']['chapters'], but
+            # I am yet to find a video having those
+
+        self._sort_formats(formats)
+
+        thumbnails = []
+        for image in metadata['images']:
+            thumbnails.append({
+                'url': image['url'],
+                'description': image['caption'],
+            })
+
+        rights = config['data']['attributes']['rights']
 
         return {
-            'id': player_info.get('VID') or video_id,
-            'title': title,
-            'description': player_info.get('VDE') or player_info.get('V7T'),
-            'upload_date': unified_strdate(upload_date_str),
-            'thumbnail': player_info.get('programImage') or player_info.get('VTU', {}).get('IUR'),
+            # XXX: data['id'] returns a different id, which varies according
+            # to the chosen language version of the Arte website; the metadata
+            # also varies accordingly
+
+            'id': video_id,  # == metadata['providerId']
+            'webpage_url': metadata.get('link', {}).get('url'),
```

Comment:
why?

Reply:
For live streams (<https://api.arte.tv/api/player/v2/config/de/LIVE>), this field contains a link to the currently running programme.